controllers/storagecluster/odfinfoconfig.go

```diff
@@ -0,0 +1,205 @@
+package storagecluster
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	operatorsv1alpha1 "github.com/operator-framework/api/pkg/operators/v1alpha1"
+	ocsv1 "github.com/red-hat-storage/ocs-operator/api/v4/v1"
+	rookCephv1 "github.com/rook/rook/pkg/apis/ceph.rook.io/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/klog/v2"
+	"reflect"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"strings"
+)
+
+type ConnectedClient struct {
+	Metadata  types.NamespacedName `json:"metadata"`
+	ClusterID string               `json:"clusterId"`
+}
+type InfoStorageCluster struct {
+	Metadata                types.NamespacedName `json:"metadata"`
+	StorageProviderEndpoint string               `json:"storageProviderEndpoint"`
+	CephClusterFSID         string               `json:"cephClusterFSID"`
+}
+
+type OdfInfoData struct {
+	OdfVersion          string             `json:"odfVersion"`
+	OdfDeploymentType   string             `json:"odfDeploymentType"`
+	Clients             []ConnectedClient  `json:"clients"`
+	StorageCluster      InfoStorageCluster `json:"storageCluster"`
+	StorageClusterCount int                `json:"storageClusterCount"`
+	StorageSystemName   string             `json:"storageSystemName"`
+	IsDROptimized       bool               `json:"isDROptimized"`
+}
+
+const (
+	RDROdfInfoKeyName            = "config.yaml"
+	RDROdfDeploymentTypeExternal = "external"
+	RDROdfDeploymentTypeInternal = "internal"
+	RookCephMonSecretName        = "rook-ceph-mon"
+	FsidKey                      = "fsid"
+	OdfOperatorNamePrefix        = "odf-operator"
+	RDROdfInfoConfigMapName      = "odf-info"
+	RDROdfInfoMapKind            = "ConfigMap"
+)
+
+type odfInfoConfig struct{}
+
+// ensureCreated ensures that a ConfigMap resource exists with its Spec in
+// the desired state.
+func (obj *odfInfoConfig) ensureCreated(r *StorageClusterReconciler, sc *ocsv1.StorageCluster) (reconcile.Result, error) {
+	odfInfo, configErr := getOdfInfoConfig(r, sc)
+	if configErr != nil {
+		return reconcile.Result{}, fmt.Errorf("failed to get ODF info config data: %w", configErr)
+	}
+	odfInfoKey := RDROdfInfoKeyName + "." + sc.Name
```

Comment:
Your suggestion is vague in this context. A .yaml file of a `StorageCluster` can imply many things, non of which are necessarily related to a configuration of the sort in question. Therefore I insist the key name includes `config.yaml` so that at the very least there's an indication of a CM key.

Reply:
Ok. in that case let's have  `<sc_namespace>/<sc_name>.config.yaml` instead of `config.yaml.<sc_name>`